sdk/eventhub/Azure.Messaging.EventHubs/tests/Core/EventPositionTests.cs

```diff
@@ -0,0 +1,221 @@
+            Assert.That((first == second), Is.False, "The == operator comparison is incorrect.");
+            Assert.That((first != second), Is.True, "The != operator comparison is incorrect.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventPosition "/>
+        ///   equality.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TheSameInclusiveFlagsAreEqual(bool isInclusive)
+        {
+            var first = EventPosition.FromSequenceNumber(234234, isInclusive);
+            var second = EventPosition.FromSequenceNumber(234234, isInclusive);
+
+            Assert.That(first.Equals((object)second), Is.True, "The default Equals comparison is incorrect.");
+            Assert.That(first.Equals(second), Is.True, "The IEquatable comparison is incorrect.");
+            Assert.That((first == second), Is.True, "The == operator comparison is incorrect.");
+            Assert.That((first != second), Is.False, "The != operator comparison is incorrect.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventPosition "/>
+        ///   equality.
+        /// </summary>
+        ///
+        [Test]
+        public void DifferentInclusiveFlagsAreNotEqual()
+        {
+            var first = EventPosition.FromSequenceNumber(234234, true);
+            var second = EventPosition.FromSequenceNumber(234234, false);
+
+            Assert.That(first.Equals((object)second), Is.False, "The default Equals comparison is incorrect.");
+            Assert.That(first.Equals(second), Is.False, "The IEquatable comparison is incorrect.");
+            Assert.That((first == second), Is.False, "The == operator comparison is incorrect.");
+            Assert.That((first != second), Is.True, "The != operator comparison is incorrect.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventPosition "/>
+        ///   equality.
+        /// </summary>
+        ///
+        [Test]
+        public void DifferentMembersAreNotEqual()
+        {
+            var first = EventPosition.FromSequenceNumber(234234);
+            var second = EventPosition.FromOffset(12);
+
+            Assert.That(first.Equals((object)second), Is.False, "The default Equals comparison is incorrect.");
+            Assert.That(first.Equals(second), Is.False, "The IEquatable comparison is incorrect.");
+            Assert.That((first == second), Is.False, "The == operator comparison is incorrect.");
+            Assert.That((first != second), Is.True, "The != operator comparison is incorrect.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventPosition.GetHashCode "/>
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void GetHashCodeReturnsDifferentValuesForDifferentMembers()
```

Comment:
Maybe we could add a `GetHashCodeReturnsSameValueForEqualMembers`?

Reply:
You're not guaranteed that using this implementation of the builder, but its been requested that we use it for consistency with other libraries.